src/snyk/common/services/learnService.ts

```diff
@@ -0,0 +1,113 @@
+import axios from 'axios';
+import { isCodeIssue, isOssIssue, OpenCommandIssueType } from '../../common/commands/types';
+import type { completeFileSuggestionType } from '../../snykCode/interfaces';
+import { OssIssueCommandArg } from '../../snykOss/views/ossVulnerabilityTreeProvider';
+import { ILog } from '../logger/interfaces';
+
+const ecosystems = [
+  'all',
+  'java',
+  'javascript',
+  'python',
+  'golang',
+  'php',
+  'cpp',
+  'csharp',
+  'kubernetes',
+  'ruby',
+  'elixir',
+  'docker',
+  'npm',
+] as const;
+type LessonEcosystem = typeof ecosystems[number];
+
+const isValidEcosystem = (potentialEcosystem: string | null | undefined): potentialEcosystem is LessonEcosystem => {
+  return ecosystems.includes(potentialEcosystem as LessonEcosystem);
+};
+
+export type Lesson = {
+  title: string;
+  lessonId: string;
+  ecosystem: LessonEcosystem;
+  url: string;
+};
+
+type PackageMapper = {
+  [key in LessonEcosystem]?: string[];
+};
+
+export interface ILearnService {
+  getLesson(options: { ecosystem: LessonEcosystem; cwes?: string[]; rules?: string[] }): Promise<Lesson | null>;
+}
+
+export class LearnService implements ILearnService {
+  private ecosystem: LessonEcosystem;
+  private cwes: string[] = [];
+
+  constructor(
+    private readonly issue: OssIssueCommandArg | completeFileSuggestionType,
+    private readonly issueType: OpenCommandIssueType,
+    private readonly logger: ILog,
+  ) {
+    this.logger = logger;
+    if (isCodeIssue(issue, issueType)) {
+      this.cwes = issue?.cwe || [];
+      this.ecosystem = LearnService.convertCodeIdToEcosystem(issue.id);
+    } else if (isOssIssue(issue, issueType)) {
+      this.cwes = issue?.identifiers?.CWE || [];
+      this.ecosystem = LearnService.convertOSSProjectTypeToEcosystem(issue.packageManager);
+    } else {
+      logger.error(`Issue type "${issueType}" not supported`);
```

Comment:
let's not log logging with variables is hard to look for, instead, let's look at a hard coded msg and add a payload with the needed details.
something like:
`
logger.error('Issue type  not supported',{issueType,"team":'owl'});
`

Reply:
Yeah I also wasn't happy with the logging. According to the TS the logger only takes a string. I'll look deeper to see if there is a way we can put more info & tag our team.

Reply:
yeah, the logger is expecting a string. 
```export interface ILog {
  info(message: string): void;
  warn(message: string): void;
  error(message: string): void;
  debug(message: string): void;

  log(level: LogLevel, message: string): void;

  showOutput(): void;
}```

It goes to vscode.window.createOutputChannel(SNYK_NAME).appendLine(`[${level}] ${message}`);
So I don't think it ends up in our logs... Are these the logs we show to the user @michelkaporin ?

Reply:
@denis-snyk Correct, these are the logs that gets shown in VS Code output channel. None of them reach the backend.

<img width="1260" alt="Screenshot 2022-04-01 at 11 12 50" src="https://user-images.githubusercontent.com/2239563/161233436-57c623ef-cf8e-4b02-9da3-c0caa60105f9.png">

Reply:
Since it's user facing, let's report this as `debug`, which would log it to console, instead of the output channel.

Reply:
In order to handler errors with unknown types, use the `ErrorHandler` class